pandas/core/indexes/period.py

```diff
@@ -567,17 +567,32 @@ def get_loc(self, key, method=None, tolerance=None):
         """
 
         if isinstance(key, str):
-            try:
-                return self._get_string_slice(key)
-            except (TypeError, KeyError, ValueError, OverflowError):
-                pass
 
             try:
-                asdt, reso = parse_time_string(key, self.freq)
-                key = asdt
-            except DateParseError:
-                # A string with invalid format
-                raise KeyError(f"Cannot interpret '{key}' as period")
+                loc = self._get_string_slice(key)
+                return loc
+            except (TypeError, ValueError):
```

Comment:
OverflowError can no longer occur?

Reply:
ill see if i can cook up a test case that overflows